x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_logic.ts

```diff
@@ -113,7 +117,14 @@ export const EnginesListLogic = kea<MakeLogicType<EngineListValues, EnginesListA
         openDeleteEngineModal: () => true,
       },
     ],
-
+    isFirstRequest: [
+      true,
+      {
+        apiError: () => false,
+        apiSuccess: () => false,
+        setIsFirstRequest: () => true,
+      },
+    ],
```

Comment:
I think this would be more accurate:
```suggestion
    isFirstRequest: [
      null,
      {
        makeRequest: (state) => state === null ? true : false,
      },
    ],
```